plugins/bufferlines/barbecue.nix

```diff
@@ -0,0 +1,172 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.barbecue;
+  helpers = import ../helpers.nix {inherit lib;};
+  mkListStr = helpers.defaultNullOpts.mkNullable (types.listOf types.str);
+in {
+  options.plugins.barbecue =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "barbecue-nvim";
+
+      package = helpers.mkPackageOption "barbecue-nvim" pkgs.vimPlugins.barbecue-nvim;
+
+      attachNavic = helpers.defaultNullOpts.mkBool true ''
+        Whether to attach navic to language servers automatically.
+      '';
+
+      createAutocmd = helpers.defaultNullOpts.mkBool true ''
+        Whether to create winbar updater autocmd.
+      '';
+
+      includeBuftypes = mkListStr ''[ "" ]'' ''
+        Buftypes to enable winbar in.
+      '';
+
+      excludeFiletypes = mkListStr ''[ "netrw" "toggleterm" ]'' ''
+        Filetypes not to enable winbar in.
+      '';
+
+      modifiers = {
+        dirname = helpers.defaultNullOpts.mkStr ":~:." ''
+          Filename modifiers applied to dirname.
+
+          See: `:help filename-modifiers`
+        '';
+
+        basename = helpers.defaultNullOpts.mkStr "" ''
+          Filename modifiers applied to basename.
+
+          See: `:help filename-modifiers`
+        '';
+      };
+
+      showDirname = helpers.defaultNullOpts.mkBool true ''
+        Whether to display path to file.
+      '';
+
+      showBasename = helpers.defaultNullOpts.mkBool true ''
+        Whether to display file name.
+      '';
+
+      showModified = helpers.defaultNullOpts.mkBool false ''
+        Whether to replace file icon with the modified symbol when buffer is modified.
+      '';
+
+      modified = helpers.defaultNullOpts.mkStr ''function(bufnr) return vim.bo[bufnr].modified end,'' ''
```

Comment:
Not sure the best type for this lua function, I saw some other packages doing this!

Reply:
I guess this is also whats throwing an error in CI (with the `lead_custom_section`. Thanks.

Reply:
->Question: When to choose between an `types.str` and a `helpers.rawType` option for lua code ?

- If the upstream option expects only lua code, we use `types.str`. There is no ambiguity and the user doesn't have to bother with providing a `rawType` value.
In this case, we do the conversion ourselves in the `config` section of the plugin: `foo = helpers.ifNonNull' foo (helpers.mkRaw foo);`
- If the upstream option accept both strings and lua code (or when there is some kind of ambiguity), we rely on `helpers.rawType`. In this case, there is no need to convert the provided value.

Reply:
Okay just to replay it back to you; we carry on using the str type. I just need to update the config section with this helpers function (?) you defined.